siad/renter.go

```diff
@@ -33,6 +41,22 @@ func (d *daemon) renterDownloadHandler(w http.ResponseWriter, req *http.Request)
 	writeSuccess(w)
 }
 
+// renterDownloadqueueHandler handles the API call to request the download
+// queue.
+func (d *daemon) renterDownloadqueueHandler(w http.ResponseWriter, req *http.Request) {
+	downloads := d.renter.DownloadQueue()
+	downloadSet := make([]DownloadInfo, 0, len(downloads))
+	for _, dl := range downloads {
+		downloadSet = append(downloadSet, DownloadInfo{
+			Completed:   dl.Completed(),
+			Destination: dl.Destination(),
+			Nickname:    dl.Nickname(),
+		})
```

Comment:
I'm not really seeing the value of an interface if all we do is immediately call all of its getters...
It'd be another thing if we kept the `downloadSet` around and called its getters multiple times.